lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,6 +193,11 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
+        if user.is_anonymous():
```

Comment:
I kept the is_anonymous() check here, but created a SystemUser subclass of AnonymousUser, and added a check for that subclass, so our user doesn't have to be un`known`.